src/preamble/01_what_do_we_mean_when_we_say_cabal.md

```diff
@@ -1 +1,96 @@
+   creating packages and building their contents. This build system is used by
+   cabal-install and other package managers, stack for example.
+
+3. [cabal-install](#the-binary-cabal-install-cli-tool) is the command line
+   utility to help configure, compile and install Haskell libraries and
+   programs.
+
+## The .cabal file
+
+The .cabal file contains information that drives the compilation and building of
+Haskell packages. The .cabal file lives at the root directory that contains the
+haskell source code corresponding to the package. By convention this file is
+named `<project-name>.cabal`. It is a text-based, key-value format, that is
+divided into subsections called stanzas.
+
+The package properties describe the package as a whole, such as name, license,
+author, dependenices, etc. It also contains information about optional
+components such as [library](../new_to_cabal/06_first_cabal_library.md),
+[executables](../new_to_cabal/07_first_cabal_executable.md),
+[test-suite](../leveling_up/02_first_cabal_test-suite.md),
+[benchmark](src/leveling_up/03_first_cabal_benchmark.md). These components are
+also called stanzas.
+
+One of the purposes of Cabal is to make it easier to build a package with
+different Haskell implementations. So it provides abstractions of features
+present in different Haskell implementations and wherever possible it is best to
+take advantage of these to increase portability. For example one of the pieces
+of information an author can put in the package’s .cabal file is what language
+extensions the code uses. This is preferable to specifying flags for a specific
+compiler as it allows Cabal to pick the right flags for the Haskell
+implementation that the user picks. It also allows Cabal to figure out if the
+language extension is even supported by the Haskell implementation that the user
+picks. Where compiler-specific options are needed, there is an “escape hatch”
+available. The developer can specify implementation-specific options and more
+generally there is a configuration mechanism to customise many aspects of how a
+package is built depending on the Haskell implementation, the operating system,
+computer architecture and user-specified configuration flags.
+
+There are various
+[Haskell implementations](https://wiki.haskell.org/Implementations), but we have
+now pretty much all converged on [GHC](https://www.haskell.org/ghc/) as the
+standard Haskell implementation.
+
+## The CABAL library
+
+Cabal stands for Common Architecture for Building Applications and Libraries.
+This is the library that provides functionality that allows the information in
+the .cabal files to be put to use. The Cabal library contains the code for
+parsing .cabal files and operations for building haskell packages. Cabal the
+library, by convention, is written with Capitalization case. Cabal can take a
+haskell dependency graph (of external and internal modules) and use GHC to build
+it.
+
+## The binary cabal-install (cli tool)
```

Comment:
This is hard, because the exe is not called `cabal-install` and we don't want to mislead the users. How about "The commandline tool (cabal-instal)"? I'm not happy with that either.